File: cyberheart.c
#include <stdio.h>
#include <string.h>

void printHeart(char *str) {
    int n = strlen(str);
    int i, j;

    for (i = n / 2; i <= n; i += 2) {
        for (j = 1; j < n - i; j += 2)
            printf(" ");
        for (j = 1; j <= i; j++)
            printf("%c", str[j % n]);
        for (j = 1; j <= n - i; j++)
            printf(" ");
        for (j = 1; j <= i; j++)
            printf("%c", str[j % n]);
        printf("\n");
    }

    for (i = n; i >= 1; i--) {
        for (j = i; j < n; j++)
            printf(" ");
        for (j = 1; j <= (i * 2) - 1; j++)
            printf("%c", str[j % n]);
        printf("\n");
    }
}

int main() {
    char str[] = " i love cybersecurity";
    printHeart(str);
    return 0;
}

File: cyberheart.py
def print_heart(s):
    n = len(s)
    
    # Parte superior do coração
    for i in range(n // 2, n + 1, 2):
        # Espaços à esquerda
        for j in range(1, n - i, 2):
            print(" ", end="")
        
        # Primeira metade do coração
        for j in range(1, i + 1):
            print(s[j % n], end="")
        
        # Espaços entre as duas metades
        for j in range(1, n - i + 1):
            print(" ", end="")
        
        # Segunda metade do coração
        for j in range(1, i + 1):
            print(s[j % n], end="")
        
        print()
    
    # Parte inferior do coração
    for i in range(n, 0, -1):
        # Espaços à esquerda
        for j in range(i, n):
            print(" ", end="")
        
        # Parte inferior do coração (formato invertido)
        for j in range(1, (i * 2) - 1 + 1):
            print(s[j % n], end="")
        
        print()

# String a ser exibida no coração
str = " i love cybersecurity"
print_heart(str)
